scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -498,6 +521,13 @@ object Execution {
   private case class WriteExecution[T](head: ToWrite, tail: List[ToWrite], fn: (Config, Mode) => T) extends Execution[T] {
 
     /**
+     * override this so that we can push down the conf value to the current TypedPipes
+     */
+    override def setExtraConfig(conf: Config): Unit = {
+      (head :: tail).foreach(_.setExtraConfig(conf))
```

Comment:
This is only setting step configs on a single step for each ToWrite, that won't set the config options globally for a given run. Its not really clear what it would mean since it won't apply for a full execution.